#### Hyphotesis Test_1

##### Set the hypothesis

* H0: Test group completion rate (CR) = Control group completion rate (CR)
* H1: Test group completion rate (CR) != Control group completion rate (CR)

* significance level = 0.05

In [83]:
# import necessary library
from scipy import stats

### z-test (for comparing proportions) 

In [84]:
# Completion rates and sample sizes for test and control groups
completion_rate_test = 0.14
completion_rate_control = 0.12
n_test = 176641
n_control = 140482

# Calculate sample proportions
p1 = completion_rate_test
p2 = completion_rate_control

# Calculate pooled proportion
p = (p1 * n_test + p2 * n_control) / (n_test + n_control)

# Compute standard error
SE = np.sqrt(p * (1 - p) * (1 / n_test + 1 / n_control))

# Compute z-statistic
Z = (p1 - p2) / SE

# Calculate p-value
p_value = 2 * (1 - stats.norm.cdf(np.abs(Z)))  # Two-tailed test

# Print results
print("Z-statistic:", Z)
print("p-value:", p_value)


Z-statistic: 16.5740796708499
p-value: 0.0


##### Decision

Since p-value (p-value: 0.0) is less than the choosen significance level, we reject the null hypothesis. And , conclude that:  Test group completion rate (CR) is not equal to Control group completion rate (CR).

#### Hyphotesis Test_2

##### Set the hypothesis

* H0: All clients in test group average/mean total process time > All clients in control group average/mean total process time
* H1: All clients in test group average/mean total process time < All clients in control group average/mean total process time

* significance level = 0.05

In [126]:
# Load the data samples
sample1 = pd.read_csv("time_differ1_test.csv")
sample2 = pd.read_csv("time_differ1_control.csv")
# Change the datatype to timedelta to both group
sample1['time_diffrence'] = pd.to_timedelta(sample1['time_diffrence'])
sample2['time_diffrence'] = pd.to_timedelta(sample2['time_diffrence'])

In [134]:
# Cahnge timedelta to integer dataype to test group (We use the .dt.total_seconds() method to convert the timedeltas to total seconds.
# Then, we use .astype(int) to convert the total seconds to integers.)
sample1['time_diffrence'] = sample1['time_diffrence'].dt.total_seconds().astype(int)

In [136]:
# Cahnge timedelta to integer dataype to control group
sample2['time_diffrence'] = sample2['time_diffrence'].dt.total_seconds().astype(int)

### Two sample T-test

In [139]:
t_statistic, p_value = stats.ttest_ind(sample1, sample2, equal_var=False)

# Print the t-statistic and p-value
print("t-statistic:", t_statistic)
print("p-value:", p_value)

t-statistic: [-3.50045587]
p-value: [0.00046454]


##### Decision

* Total time spent: (control group) : '1363 days 14:54:24'
* Total time spent: (test Group) : '1624 days 05:10:41'
* Average time spent: (control Group) : '0 days 00:16:47.360580047'
* Average time spent: (test_Group): '0 days 00:15:37.548376536'

Since p-value (p-value: [0.00046454] is less than the choosen significance level (Alpha : 0.05), we reject the null hypothesis. And , conclude that:  All clients in test group average/mean total process time is shorter or less than to All clients in control group average/mean total process time.

#### Hyphotesis Test_3
* significance level = 0.05
##### Hyphotesis Test_3.1 (Control group)

##### Set the hypothesis

* H0: For the clients in the control group , is the mean time spent in each step = among all the steps.
* H1: For the clients in the control group , is the mean time spent in each step != among all the steps.

In [170]:
# load the dataset for control group
control_group = pd.read_csv("control_avg_time_spent_per_step.csv")
control_group.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 92093 entries, 0 to 92092
Data columns (total 1 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   mean_time_each_step_in_sec  92093 non-null  float64
dtypes: float64(1)
memory usage: 719.6 KB


In [171]:
C_mean_if_same_across_the_step = 92093 * 6
meanOf_C_mean_if_same_across_the_step = C_mean_if_same_across_the_step/92093
meanOf_C_mean_if_same_across_the_step

6.0

#### Python way

In [172]:
st.ttest_1samp(control_group, 6.0)

TtestResult(statistic=array([25.31751889]), pvalue=array([6.23962297e-141]), df=array([92092]))

We found extremly small p-value so, we reject the null-hypothesis.

##### Hyphotesis Test_3.2 (Test group)

##### Set the hypothesis

* H0: For the clients in the test group , is the mean time spent in each step = among all the steps.
* H1: For the clients in the test group , is the mean time spent in each step != among all the steps.

In [168]:
# Load the dataset for test  group
test_group = pd.read_csv("test_avg_time_spent_per_step.csv")
test_group.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110607 entries, 0 to 110606
Data columns (total 1 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   mean_time_each_step_in_sec  110607 non-null  float64
dtypes: float64(1)
memory usage: 864.2 KB


In [173]:
T_mean_if_same_across_the_step = 110607 * 30
meanOf_T_mean_if_same_across_the_step = T_mean_if_same_across_the_step/110607
meanOf_T_mean_if_same_across_the_step

30.0

### Python way

In [174]:
st.ttest_1samp(control_group, 30)

TtestResult(statistic=array([25.25339417]), pvalue=array([3.13017497e-140]), df=array([92092]))

Here also we found extremly small p-value so, we reject the null-hypothesis.

#### Hyphotesis Test_4

##### Set the hypothesis


* H0: The mean time in each step in control group = mean time in each step in test group.
* H1: The mean time in each step in control group != mean time in each step in test group.

* significance level = 0.05

### Two sample T-test

In [175]:
st.ttest_ind(test_group,control_group, equal_var=False)

TtestResult(statistic=array([-6.642999]), pvalue=array([3.0825866e-11]), df=array([174979.41839847]))

Finally, in this case also we found a very low p-value which is 0.000000000030825866, implies very low number than the alpha value so, we reject the null hypotheis.